### PR TITLE
Const optimize

### PR DESCRIPTION
**Pull request by sander-visser**

### CODE REVIEW

File: apx/common/src/apx_nodeInfo.c [hunk 1]
```diff
@@ -28,6 +28,7 @@ static void apx_nodeInfo_connectRequirePortInternal(apx_nodeInfo_t *self, int32_
 static void apx_nodeInfo_connectProvidePortInternal(apx_nodeInfo_t *self, int32_t portIndex, apx_node_t *requesterNode, apx_port_t *requirePort);
 static void apx_nodeInfo_disconnectRequirePortInternal(apx_nodeInfo_t *requesterNodeInfo, int32_t requesterPortIndex);
 static void apx_nodeInfo_disconnectProvidePortInternal(apx_nodeInfo_t *providerNodeInfo, int32_t providerPortIndex, apx_portref_t *portref);
+static inline bool apx_nodeInfo_isPortEntryOutside(const apx_portDataMapEntry_t* portEntry, uint32_t portDataLen);
```
Comment (cogu) [inline apx/common/src/apx_nodeInfo.c]: According to MSDN documentation the inline keyword is only available in c++ compiler. I would remove "inline" for now until we have tested all versions of visual studio C compilers (2013, 2015, 2017)

File: apx/common/src/apx_nodeInfo.c [hunk 2]
```diff
@@ -461,34 +462,31 @@ void apx_nodeInfo_copyInitDataFromProvideConnectors(apx_nodeInfo_t *self)
...
-            assert(requirePortEntry != 0);
+            
             providePortIndex = portref->port->portIndex;
             assert( (provideNodeInfo != 0) && (providePortIndex>=0) );
             provideNodeData = provideNodeInfo->nodeData;
             if ( (provideNodeData!=0) && (provideNodeData->outPortDataBuf != 0) )
             {
-               apx_portDataMapEntry_t *providePortEntry;
-               providePortEntry = apx_portDataMap_getEntry(&provideNodeInfo->outDataMap, providePortIndex);
+               const apx_portDataMapEntry_t* const providePortEntry = apx_portDataMap_getEntry(&provideNodeInfo->outDataMap, providePortIndex);
```
Comment (cogu) [inline apx/common/src/apx_nodeInfo.c:478]: All variables must be declared in beginning of scope, before first statement. This is to make it possible to compile in C89 compilers such as VS2010

File: apx/common/src/apx_nodeInfo.c [hunk 3]
```diff
@@ -461,34 +462,31 @@ void apx_nodeInfo_copyInitDataFromProvideConnectors(apx_nodeInfo_t *self)
...
                assert(providePortEntry != 0);
                //array bounds check
-               if ( ( (uint32_t)providePortEntry->offset >= provideNodeData->outPortDataLen) ||  ( ((uint32_t)(providePortEntry->offset+providePortEntry->length)) > provideNodeData->outPortDataLen) )
+               if ( apx_nodeInfo_isPortEntryOutside(providePortEntry, provideNodeData->outPortDataLen))
```
Comment (cogu) [inline apx/common/src/apx_nodeInfo.c]: Feels like this function needs a better name, it is outside what?
perhaps apx_nodeInfo_isPortEntryOutsideLen

File: apx/common/src/apx_nodeInfo.c [hunk 4]
```diff
@@ -601,7 +604,7 @@ static void apx_nodeInfo_connectProvidePortInternal(apx_nodeInfo_t *self, int32_
             end = adt_ary_length(innerConnectionList);
             for(i=0;i<end;i++)
             {
-               apx_portref_t *ref = (apx_portref_t*) *adt_ary_get(innerConnectionList,i);
+               apx_portref_t *ref = (apx_portref_t*) *adt_ary_get(innerConnectionList,i); // todo adt_ary_value
```
Comment (cogu) [inline apx/common/src/apx_nodeInfo.c]: If you write ///TODO: some IDE:s will be able to list TODOs much easier

File: apx/common/src/apx_nodeInfo.c (hunk 1 above)
Comment (sander-visser) [reply apx/common/src/apx_nodeInfo.c]: visual studio seems to be c90 that does not ahve inline support according to:
https://en.wikipedia.org/wiki/Inline_function#Standard_support

should we use a "#ifdef _MSC_VER" and __inline extension that seems to be supported by all modern visual studio versions? Or should we just skip inline alltogether...

Comment (sander-visser) [reply apx/common/src/apx_nodeInfo.c]: removed the inline for now to support all compilers

**Review by cogu — APPROVED** [review]
I was thinking that we can define the macro INLINE as "__inline" for Visual Studio and "inline" for gcc and then use the INLINE macro wherever we need it. Perhaps even put INLINE definition in a common header like apx_types.h. But this is an improvement we can do later.

File: apx/common/src/apx_nodeInfo.c (hunk 2 above)
Comment (sander-visser) [reply apx/common/src/apx_nodeInfo.c:478]: will update the pull request

File: apx/common/src/apx_nodeInfo.c (hunk 4 above)
Comment (sander-visser) [reply apx/common/src/apx_nodeInfo.c]: will update so TODO isnt needed - was a reminder sto self that I must have missed when committing...

File: apx/common/src/apx_nodeInfo.c (hunk 2 above)
Comment (sander-visser) [reply apx/common/src/apx_nodeInfo.c:478]: actually it is at the beginning of the if scope since 2 rows replaced by one...

**Comment by sander-visser** [comment]
should now be ready for merge - will leave the repo until your branch is merged - then I will look into inlining after having profiled the latest version built with O2 by gcc...

**Review by cogu — APPROVED** [review]

**Pull request merged**